dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java

```diff
@@ -167,16 +152,36 @@ private void initContextMenu() {
         createBucket.setOnAction(event -> createBucketPrompt());
         createFolder = new MenuItem(resourceBundle.getString("createFolderContextMenu"));
         createFolder.setOnAction(event -> createFolderPrompt());
-        contextMenu.getItems().addAll(metaData, physicalPlacement, deleteFile, deleteFolder, deleteBucket, new SeparatorMenuItem(), createBucket, createFolder);
+        contextMenu.getItems().addAll(metaData, physicalPlacement, deleteFile, deleteFolder, deleteBucket, selectAll, new SeparatorMenuItem(), createBucket, createFolder);
+    }
+
+    private void selectAll() {
+        final List<String> rowNameList = new ArrayList<>();
```

Comment:
Replace this with a set.  This will improve the runtime of the check on line 164 from O(n) to constant time.